quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -168,6 +175,19 @@ impl TypedSourceFactory for FileSourceFactory {
     }
 }
 
+pub fn dir_and_filename(filepath: &Path) -> anyhow::Result<(Uri, &Path)> {
+    let dir_uri: Uri = filepath
+        .parent()
+        .ok_or_else(|| anyhow::anyhow!("Parent directory could not be resolved"))?
```

Comment:
you need an import from anyhow... I forgot the name of the trait.


```suggestion
        .context("Parent directory could not be resolved")?
```

Reply:
I actually found quite a few other of these in the codebase:
![image](https://github.com/quickwit-oss/quickwit/assets/7913347/c52393e5-bfe0-443d-9a9d-16f9507481f0)

But I would prefer not to touch unrelated pieces of code in the context of this PR. Fine by you?

Reply:
Yes you can ignore them